### DISCUSSION TITLE
Is lib really required?

### DISCUSSION THREAD

**Discussion by Jelmerta** [sotrh/learn-wgpu#575] [Q&A]
Hi, I'm wondering if it is really required to have a definition for lib in Cargo.toml as mentioned here:
https://github.com/sotrh/learn-wgpu/tree/master/docs/beginner/tutorial1-window#added-support-for-the-web

```
[lib]
crate-type = ["cdylib", "rlib"]
```

I have been able to build my application (https://github.com/jelmerta/kloenk) both as a desktop application as well as running in the web without having this definition. wasm-pack does seem to require it, perhaps because it builds more bindings between js and rust? However, `cargo run` runs just fine. `trunk serve` also works just fine.

**Comment by Jelmerta** [comment]
also i got winit 0.30 working for both desktop and web, based on a wgpu pull request, but it perhaps doesn't deserve a beauty award

**Comment by sotrh** [comment]
Your correct, the lib requirement is mostly for wasm-pack.
